bioconda_utils/utils.py

```diff
@@ -1193,8 +1197,12 @@ class RepoData:
     columns = _load_columns + ['channel', 'subdir', 'platform']
     #: Platforms loaded
     platforms = ['linux', 'osx', 'noarch']
-    #: Channels loaded
-    channels = ['conda-forge', 'bioconda', 'defaults']
+    # config object
+    config = None
+
+    @classmethod
+    def register_config(cls, config):
+        cls.config = config
```

Comment:
Should check that `__instance == None` to ensure the config is set before the singleton is instantiated.

Reply:
Also consider `set_cache()`:

 - A change in channels invalidates the cache.
 - A single method setting both channels and cache is probably cleaner.
 - It may be easier to have a default for the channels than to set it everywhere.